src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,85 @@
     private static final Injector injector = new Injector();
+    private static final String CANT_CREATE_AN_OBJECT = "Can't create an object Clazz + ";
+    private static final String CANT_CREATE_AN_FIELD = "Can't initialize field value + Clazz ";
+    private static final String CANT_FIND_COMP = "Injection failed. Missing "
+            + "@Component annotation on the class: ";
+    private static final String NO_FOUND_IMPL = "No implementation "
+            + "found for interface: ";
+
+    private final Map<Class<?>, Class<?>> interfaceImplementations = Map.of(
+            FileReaderService.class, FileReaderServiceImpl.class,
+            ProductParser.class, ProductParserImpl.class,
+            ProductService.class, ProductServiceImpl.class);
+
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = interfaceClazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            if (field.isAnnotationPresent(Inject.class)) {
+                Object fieldInstance = getInstance(field.getType());
+                clazzImplementationInstance = createNewInstance(clazz);
+
+                try {
+                    field.setAccessible(true);
+                    field.set(clazzImplementationInstance, fieldInstance);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(CANT_CREATE_AN_FIELD + clazz.getName()
+                            + " Field " + field.getType());
+                }
+            }
+        }
+        if (clazzImplementationInstance == null) {
+            clazzImplementationInstance = createNewInstance(clazz);
+        }
+        return clazzImplementationInstance;
+    }
+
+    private Object createNewInstance(Class<?> clazz) {
+        if (!clazz.isAnnotationPresent(Component.class)) {
+            throw new RuntimeException(
+                    CANT_FIND_COMP + clazz.getName());
+        }
+
+        if (instances.containsKey(clazz)) {
+            return instances.get(clazz);
+        }
+
+        try {
+            Constructor<?> constructor = clazz.getConstructor();
+            Object instance = constructor.newInstance();
+            instances.put(clazz, instance);
+            return instance;
+        } catch (ReflectiveOperationException e) {
+            throw new RuntimeException(CANT_CREATE_AN_OBJECT + clazz.getName());
+        }
+    }
+
+    private Class<?> findImplementation(Class<?> interfaceClazz) {
+        if (interfaceClazz.isInterface()) {
+            return interfaceImplementations.get(interfaceClazz);
+        } else {
+            throw new RuntimeException(NO_FOUND_IMPL + interfaceClazz.getName());
+        }
```

Comment:
u dont need else clause here
```suggestion
        if (interfaceClazz.isInterface()) {
            return interfaceImplementations.get(interfaceClazz);
        } 
        throw new RuntimeException(NO_FOUND_IMPL + interfaceClazz.getName());
```